src/features/vault.ts

```diff
@@ -0,0 +1,173 @@
+import * as vscode from 'vscode';
+import * as utilAnsibleCfg from './utils/ansibleCfg';
+import * as tmp from 'tmp';
+import * as cp from 'child_process';
+import * as fs from "fs";
+
+async function askForVaultId(ansibleCfg: string) {
+  let vaultId: string = "default";
+  if (ansibleCfg === "ANSIBLE_VAULT_IDENTITY_LIST") {
+    let envVar : any = process.env.ANSIBLE_VAULT_IDENTITY_LIST;
+    var identityList = envVar.split(',').map( (id : string) => id.split('@', 2)[0].trim());
+  } else {
+    let cfg : any = utilAnsibleCfg.getValueByCfg(ansibleCfg);
+    var identityList = cfg.defaults.vaultIdentity_list.split(',').map( (id : string) => id.split('@', 2)[0].trim());
+  }
+  await vscode.window.showQuickPick(identityList)
+    .then( chosen => { if (chosen) { vaultId = chosen; } } );
+  return vaultId;
+}
+
+export var toggleEncrypt = async () => {
+	let editor = vscode.window.activeTextEditor;
+	if (!editor) {
+		return;
+	}
+
+	let selection = editor.selection;
+	if (!selection) {
+		return;
+	}
+
+	let config = vscode.workspace.getConfiguration('ansible.vault');
+	let doc = editor.document;
+
+	// Read `ansible.cfg` or environment variable
+	let rootPath = utilAnsibleCfg.getRootPath(editor.document.uri);
+	if (!!process.env.ANSIBLE_VAULT_IDENTITY_LIST) {
+		var keyInCfg = "ANSIBLE_VAULT_IDENTITY_LIST";
```

Comment:
This should be defined above and set here. There is no type safety this way. It is also better to use `let` to avoid hoisting.

Reply:
fixed in c1be20b